web_pwa_cache/static/src/js/worker/core/osv/expression.js

```diff
@@ -0,0 +1,1723 @@
+                    } else {
+                        params = [];
+                        for (const it of right) {
+                            if (it) {
+                                params.push(it);
+                            }
+                        }
+                        check_null = params.length < right.length;
+                    }
+                    if (_.isEmpty(params)) {
+                        // The case for (left, 'in', []) or (left, 'not in', []).
+                        query = operator === "in" ? "FALSE" : "TRUE";
+                    } else {
+                        let instr = "";
+                        if (left === "id") {
+                            instr = new Array(params.length).fill("?").join(",");
+                        } else {
+                            const field = model.fields[left];
+                            instr = new Array(params.length)
+                                .fill(field.column_format || "?")
+                                .join(",");
+                            params = params.map(x => convert_to_column(field, x));
+                        }
+                        query = `(${table_alias}."${left}" ${operator} (${instr}))`;
+                    }
+                    if (
+                        (operator === "in" && check_null) ||
+                        (operator === "not in" && !check_null)
+                    ) {
+                        query = `(${query} OR ${table_alias}."${left}" IS NULL)`;
+                    } else if (operator === "not in" && check_null) {
+                        // Needed only for TRUE.
+                        query = `(${query} AND ${table_alias}."${left}" IS NOT NULL)`;
+                    }
+                } else {
+                    // Must not happen
+                    throw Error(`Invalid domain term ${leaf}`);
+                }
+            } else if (
+                left in model.fields &&
+                model.fields[left].type === "boolean" &&
+                ((operator === "=" && !right) || (operator === "!=" && right))
+            ) {
+                query = `(${table_alias}."${left}" IS NULL or ${table_alias}."${left}" = false )`;
+                params = [];
+            } else if (!right && operator === "=") {
+                query = `${table_alias}."${left}" IS NULL `;
+                params = [];
+            } else if (
+                left in model.fields &&
+                model.fields[left].type === "boolean" &&
+                ((operator === "!=" && !right) || (operator === "==" && right))
+            ) {
+                query = `(${table_alias}."${left}" IS NOT NULL and ${table_alias}."${left}" != 0)`;
+                params = [];
+            } else if (!right && operator === "!=") {
+                query = `${table_alias}."${left}" IS NOT NULL`;
+                params = [];
+            } else if (operator === "=?") {
+                if (right) {
```

Comment:
Think I spotted a bug here - should be `(!right)`? Popped up when doing a `name_search` on `state` on partner form view, while having a country selected.

Reply:
Fixed! thanks!!